### ISSUE TITLE
[BUG] [PostgreSQL] Upgrade fails when there is user-defined tablespace

### ISSUE THREAD

**Issue by to-bar** [epiphany-platform/epiphany#2573]
**Describe the bug**
Upgrade fails when there is user-defined tablespace:
```
TASK [postgresql : PostgreSQL | Extensions | repmgr | Clone standby node with repmgr] ********************************************************************************
fatal: [pg2]: FAILED! => {"changed": false, "cmd": ["/usr/bin/repmgr", "standby", "clone", "-h", "pg1", "-U", "epi_repmgr_admin", "-d", "epi_repmgr", "-p", "5432", "--force"], "delta": "0:00:01.677081", "end": "2021-09-10 15:31:57.557146", "msg": "non-zero return code", "rc": 14, "start": "2021-09-10 15:31:55.880065", "stderr": "NOTICE: destination directory \"/var/lib/postgresql/13/main\" provided\nINFO: connecting to source node\nDETAIL: connection string is: host=pg1 user=epi_repmgr_admin port=5432 dbname=epi_repmgr\nDETAIL: current installation size is 31 MB\nINFO: replication slot usage not requested;  no replication slot will be set up for this standby\nNOTICE: checking for available walsenders on the source node (2 required)\nNOTICE: checking replication connections can be made to the source server (2 required)\nWARNING: data checksums are not enabled and \"wal_log_hints\" is \"off\"\nDETAIL: pg_rewind requires \"wal_log_hints\" to be enabled\nWARNING: directory \"/var/lib/postgresql/13/main\" exists but is not empty\nNOTICE: -F/--force provided - deleting existing data directory \"/var/lib/postgresql/13/main\"\nNOTICE: starting backup (using pg_basebackup)...\nHINT: this may take some time; consider using the -c/--fast-checkpoint option\nINFO: executing:\n  /usr/lib/postgresql/13/bin/pg_basebackup -l \"repmgr base backup\"  -D /var/lib/postgresql/13/main -h pg1 -p 5432 -U epi_repmgr -X stream \npg_basebackup: error: directory \"/tablespace\" exists but is not empty\npg_basebackup: removing contents of data directory \"/var/lib/postgresql/13/main\"\nERROR: unable to take a base backup of the source server\nHINT: data directory (\"/var/lib/postgresql/13/main\") may need to be cleaned up manually", "stderr_lines": ["NOTICE: destination directory \"/var/lib/postgresql/13/main\" provided", "INFO: connecting to source node", "DETAIL: connection string is: host=pg1 user=epi_repmgr_admin port=5432 dbname=epi_repmgr", "DETAIL: current installation size is 31 MB", "INFO: replication slot usage not requested;  no replication slot will be set up for this standby", "NOTICE: checking for available walsenders on the source node (2 required)", "NOTICE: checking replication connections can be made to the source server (2 required)", "WARNING: data checksums are not enabled and \"wal_log_hints\" is \"off\"", "DETAIL: pg_rewind requires \"wal_log_hints\" to be enabled", "WARNING: directory \"/var/lib/postgresql/13/main\" exists but is not empty", "NOTICE: -F/--force provided - deleting existing data directory \"/var/lib/postgresql/13/main\"", "NOTICE: starting backup (using pg_basebackup)...", "HINT: this may take some time; consider using the -c/--fast-checkpoint option", "INFO: executing:", "  /usr/lib/postgresql/13/bin/pg_basebackup -l \"repmgr base backup\"  -D /var/lib/postgresql/13/main -h pg1 -p 5432 -U epi_repmgr -X stream ", "pg_basebackup: error: directory \"/tablespace\" exists but is not empty", "pg_basebackup: removing contents of data directory \"/var/lib/postgresql/13/main\"", "ERROR: unable to take a base backup of the source server", "HINT: data directory (\"/var/lib/postgresql/13/main\") may need to be cleaned up manually"], "stdout": "", "stdout_lines": []}
```

**How to reproduce**
1. Create Epiphany cluster with 2 postgresql nodes (with replication).
2. Run on standby node:
```
sudo mkdir /tablespace
sudo chown postgres:postgres /tablespace
```
3. Run on primary node:
```
sudo mkdir /tablespace
sudo chown postgres:postgres /tablespace
sudo su - postgres
psql
CREATE TABLESPACE foo LOCATION '/tablespace';
CREATE TABLE bar (id serial, name text) TABLESPACE foo;
```

4. Run `epicli upgrade`.

**Expected behavior**
Upgrade succeeded.

**Environment**
- OS: Ubuntu 18.04

**Additional context**
https://www.postgresql.org/docs/10/sql-createtablespace.html

---

**DoD checklist**

* [ ] Changelog updated (if affected version was released)
* [x] COMPONENTS.md updated / doesn't need to be updated
* [x] Automated tests passed (QA pipelines)
  * [ ] apply
  * [x] upgrade
* [ ] Case covered by automated test (if possible)
* [ ] Idempotency tested
* [x] Documentation updated / doesn't need to be updated
* [x] All conversations in PR resolved
* [x] Backport tasks created / doesn't need to be backported

**Comment by przemyslavic** [comment]
:heavy_check_mark: The upgrade was successful with user-defined tablespaces (1 master + 1 replica).

**Comment by przemyslavic** [comment]
@atsikham 
❌ However it fails when having only 1 PostgreSQL node:
```
19:24:47 INFO cli.engine.ansible.AnsibleCommand - TASK [postgresql : PostgreSQL | Query custom tablespaces] **********************
19:24:47 ERROR cli.engine.ansible.AnsibleCommand - fatal: [ci-11to12azceflannel-postgresql-vm-0]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: ssh: Could not resolve hostname ci-11to12azceflannel-postgresql-vm-0: Name or service not known", "unreachable": true}
```
Easy to reproduce on Azure.
Moving back to `in progress`😢.

**Comment by przemyslavic** [comment]
:heavy_check_mark: Fixed in #2635. 
Tested 1-node and 2-node clustered installation (upgrade).
